include/boost/windows_sspi/context.hpp

```diff
@@ -8,12 +8,12 @@
 // file LICENSE_1_0.txt or copy at http://www.boost.org/LICENSE_1_0.txt)
 //
 
-#ifndef BOOST_ASIO_WINDOWS_SSPI_CONTEXT_HPP
-#define BOOST_ASIO_WINDOWS_SSPI_CONTEXT_HPP
+#ifndef BOOST_WINDOWS_SSPI_CONTEXT_HPP
+#define BOOST_WINDOWS_SSPI_CONTEXT_HPP
 
-#include <boost/asio/windows_sspi/context_base.hpp>
-#include <boost/asio/windows_sspi/detail/sspi_functions.hpp>
-#include <boost/asio/windows_sspi/error.hpp>
+#include <boost/windows_sspi/context_base.hpp>
+#include <boost/windows_sspi/detail/sspi_functions.hpp>
+#include <boost/windows_sspi/error.hpp>
 
 // TODO: Avoid cluttering global namespace (and avoid Windows headers if possible)
 #define SECURITY_WIN32
```

Comment:
macros belonging to your library need to be prefixed with `BOOST_WINDOWS_SSPI_`

Reply:
If I had to define a macro, I would try to avoid doing so in a header, but if I had to define a macro in a header, I would do my best to ensure that it wouldn't effect users of header and even if I felt there was no way to avoid that, I definitely wouldn't name it something this generic :-)

Anyway, this specific macro needs to be defined before including the relevant Windows headers for SSPI. The TODO is there to remind me to try and avoid including the Windows headers as they are most often very nasty and can break tons of completely unrelated code, which I of course want to avoid.

If there ever was a award for worst C header ever written, I would be surprised if `windows.h` wouldn't win that award :-)

Reply:
See https://github.com/boostorg/winapi

Reply:
> See https://github.com/boostorg/winapi

Thanks, I've looked into that already. There's a reason that exists and I might be able to use some parts of it, but unfortunately it doesn't really solve all the issues.

As an example, I would probably be able to map the error codes returned from SSPI into an enum an use that instead without too many worries, but I'm a bit more worried about redefining a struct that is returned from one of the Windows API functions as a pointer (or passed to the function as a pointer) and assume that I could just use that. I would assume that could cause some really nasty and hard to debug issues for potential users.

I have looked a bit into how other libraries (including boost libraries) are dealing with the pain of supporting Windows, but I haven't really found any silver bullet.

Ensuring that I don't define this macro (`WIN32_SECURITY`) for any users of my headers would probably be a huge step in the right direction though. It seems to mean that I want to use SSPI in a user mode application (as opposed to a kernel driver) btw.

Reply:
Yeah Boost.WinApi is not perfect. And it is of course going to be missing most of the very-specific APIs.